services/app-api/src/postgres/contractAndRates/withdrawContract.ts

```diff
@@ -151,20 +156,33 @@ const withdrawContractInsideTransaction = async (
     })
 
     if (!rates) {
-        throw new NotFoundError('Could not find rates on contract')
+        throw new NotFoundError(
+            'Could not find rates on contract to be withdrawn'
+        )
     }
 
     // Child rates to withdraw with the submission, so it includes the same withdraw reason
     const ratesToWithdraw = []
+    // Child rates that need to be reassigned a new parent contract
+    const ratesToReAssign: {
+        rateID: string
+        rateName?: string
+        reassignToContractID: string
+        contractStatus: ConsolidatedContractStatus
+    }[] = []
 
-    // loop through all rates to determine which ones can be withdrawn.
+    // Loop through all rates to determine which ones can be withdrawn or reassigned parent contract
     for (const rate of rates) {
         // latest rate revision of the current rate on the submission
         const latestRateRevision = rate.revisions[0]
-        // find the parent contract of the rate, it is the contract on the rates first submission revision
-        const firstRateRevision = rate.revisions[rate.revisions.length - 1]
+        if (!latestRateRevision.submitInfo) {
+            throw new Error(
+                `Child rate ${rate.id} of contract to be withdrawn does is not submitted.`
+            )
+        }
+
         const parentContractID =
-            firstRateRevision.submitInfo?.submittedContracts[0].contractID
+            latestRateRevision.submitInfo?.submittedContracts[0].contractID
```

Comment:
suggestion: I think this is third or fourth time I'm seeing `rateRevision.submitInfo.submittedContracts[0]` logic. This feels like good candidate for a helper - I guess its `getRelatedContracts` on rate revision

Reply:
I don't think it would be worth it yet. I have 2 instances of this and writing a new function would be way more work to try to get the types out of the query with nested selects. If I encounter this a couple more times then I'll make a function.